src/commands/ImportCommand.js

```diff
@@ -0,0 +1,104 @@
+import fs from "fs";
+import path from "path";
+import async from "async";
+import Command from "../Command";
+import progressBar from "../progressBar";
+import PromptUtilities from "../PromptUtilities";
+import ChildProcessUtilities from "../ChildProcessUtilities";
+
+export default class ImportCommand extends Command {
+  initialize(callback) {
+    const inputPath = this.input[0];
+
+    if (!inputPath) {
+      return callback(new Error("Missing argument: Path to external repository"));
+    }
+
+    const externalRepoPath = path.resolve(inputPath);
+    const externalRepoBase = path.basename(externalRepoPath);
+
+    try {
+      const stats = fs.statSync(externalRepoPath);
+      if (!stats.isDirectory()) {
+        throw new Error(`Input path "${inputPath}" is not a directory`);
+      }
+      const packageJson = path.join(externalRepoPath, "package.json");
+      const packageName = require(packageJson).name;
+      if (!packageName) {
+        throw new Error(`No package name specified in "${packageJson}"`);
+      }
+    } catch (e) {
+      if (e.code === "ENOENT") {
+        return callback(new Error(`No repository found at "${inputPath}"`));
+      }
+      return callback(e);
+    }
+
+    this.targetDir = "packages/" + externalRepoBase;
+
+    let targetDirExists;
```

Comment:
Yeah, I spent some time trying to come up with a simpler check here.  What you posted above is nicer looking, but it is actually the _inverse_ of what we want: It returns an error when the target directory does _not_ exist.